src/main/java/mSearch/tool/Version.java

```diff
@@ -0,0 +1,92 @@
+package mSearch.tool;
+
+public class Version {
+	
+	private int major;
+	private int minor;
+	private int patch;
+	
+	public Version(int major, int minor, int patch) {
+		this.major = major;
+		this.minor = minor;
+		this.patch = patch;
+	}
+	
+	public Version(String versionsstring) {
+		String[] versions = versionsstring.split("\\.");
+		if(versions.length == 3) {
+			try {
+				major = Integer.parseInt(versions[0]);
+				minor = Integer.parseInt(versions[1]);
+				patch = Integer.parseInt(versions[2]);
+			} catch(NumberFormatException ex) {
+				Log.errorLog(12344564, ex, "Fehler beim Parsen der Version '" + versionsstring + "'.");
+				major = 0;
+				minor = 0;
+				patch = 0;
+			}
+		}
+	}
+	
+	public Version() {
+		major = 0;
+		minor = 0;
+		patch = 0;
+	}
+	
+	public int getMajor() {
+		return major;
+	}
+
+	public void setMajor(int major) {
+		this.major = major;
+	}
+
+	public int getMinor() {
+		return minor;
+	}
+
+	public void setMinor(int minor) {
+		this.minor = minor;
+	}
+
+	public int getPatch() {
+		return patch;
+	}
+
+	public void setPatch(int patch) {
+		this.patch = patch;
+	}
+	
+	/**
+	 * Gibt die Version als gewichtete Zahl zurück.
+	 * @return gewichtete Zahl als Integer
+	 */
+	public int toNumber() {
+		return major * 100 + minor * 10 + patch;
+	}
+	
+	/**
+	 * Gibt die Version als String zurück
+	 * @return String mit der Version
+	 */
+	public String toString() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1161)